pkg/handlers/handlers.go

```diff
@@ -0,0 +1,36 @@
+package handlers
+
+import (
+	"io"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/sources"
+)
+
+func DefaultHandlers() []Handler {
+	return []Handler{
+		&Archive{},
+	}
+}
+
+type Handler interface {
+	FromFile(io.Reader) chan ([]byte)
+	IsFiletype(io.Reader) bool
+	New()
+}
+
+func HandleFile(file io.Reader, chunkSkel *sources.Chunk, chunksChan chan (*sources.Chunk)) bool {
```

Comment:
Is there a way to handle the filename within an archive in the metadata?

Reply:
The issue is that it would need to be added to the metadata so we'd need to include a way to handle each metadata source type. We could work around that by having the handler chunk handling be done by the source. This seemed like the best course since it makes the handlers easy to add to sources.